jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/job/Job.java

```diff
@@ -54,12 +61,39 @@
     public String getName();
 
     /**
-     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are allowed.
+     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are
+     * allowed.
      * <p>
      * Default is 2 = medium.
      *
      * @return priority
      */
     public int getPriority();
 
+
+    /**
+     * @return All involved locations
+     */
+    public List<Location> getAllLocations();
```

Comment:
I've just now understood what did you mean "Why dont we just leave the location in the corresponding activity? I think that locations do not have much meaning without its corresponding activity ..."
You are right. I'll make some notes, and maybe a prototype tomorrow! We may have to forget the builder pattern for jobs, but I would suggest this anyway, because builder pattern and inheritence is not works well together. (See JavaFX which also deprecated its builders.)